lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -126,9 +113,9 @@ protected function getFilename($id)
     {
         return $this->directory
             . DIRECTORY_SEPARATOR
-            . implode(str_split(hash('sha256', $id), 2), DIRECTORY_SEPARATOR)
+            . substr(hash('sha256', $id), 0, 2)
```

Comment:
Ah yes, it seems like ext4 (very common) is indeed not affected by that limitation, but tools start to crash and burn with over 10k files/dir